operators/src/processing/meteosat/temperature.rs

```diff
@@ -228,47 +268,28 @@ where
         ctx: &'a dyn QueryContext,
     ) -> Result<BoxStream<'a, Result<Self::Output>>> {
         let src = self.source.query(query, ctx).await?;
-
-        let rs = src.map(move |tile| {
-            let tile = tile?;
-            match &tile.grid_array {
-                GridOrEmpty::Empty(_) => Ok(RasterTile2D::new_with_properties(
-                    tile.time,
-                    tile.tile_position,
-                    tile.global_geo_transform,
-                    EmptyGrid::new(tile.grid_array.grid_shape(), OUT_NO_DATA_VALUE).into(),
-                    tile.properties,
-                )),
-                GridOrEmpty::Grid(grid) => {
-                    let lut = self.create_lookup_table(&tile)?;
-
-                    // Create result raster
-                    let mut out = Grid2D::new(
-                        grid.grid_shape(),
-                        vec![OUT_NO_DATA_VALUE; grid.number_of_elements()],
-                        Some(OUT_NO_DATA_VALUE),
-                    )
-                    .expect("raster creation must succeed");
-
-                    for (idx, &pixel) in grid.data.iter().enumerate() {
-                        if !grid.is_no_data(pixel) {
-                            let lut_idx: u64 = pixel.as_();
-                            out.data[idx] = *lut
-                                .get(lut_idx as usize)
-                                .ok_or(Error::UnsupportedRasterValue)?;
-                        }
+        let rs = src
+            .map(move |tile| {
+                tile.and_then(|t| {
+                    let satellite = self.satellite(&t)?;
+                    let channel = self.channel(&t, satellite)?;
+                    let offset = t.properties.number_property::<f64>(&self.offset_key)?;
```

Comment:
could these be determined outside of the `map`?

Reply:
no you need the tile